ec/ecresource/deploymentresource/elasticsearch/v2/schema.go

```diff
@@ -0,0 +1,479 @@
+					resource.UseStateForUnknown(),
+				},
+			},
+			"ref_id": {
+				Type:        types.StringType,
+				Description: "Optional ref_id to set on the Elasticsearch resource",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					planmodifier.DefaultValue(types.String{Value: "main-elasticsearch"}),
+					resource.UseStateForUnknown(),
+				},
+			},
+			"resource_id": {
+				Type:        types.StringType,
+				Description: "The Elasticsearch resource unique identifier",
+				Computed:    true,
+			},
+			"region": {
+				Type:        types.StringType,
+				Description: "The Elasticsearch resource region",
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.UseStateForUnknown(),
+				},
+			},
+			"cloud_id": {
+				Type:        types.StringType,
+				Description: "The encoded Elasticsearch credentials to use in Beats or Logstash",
+				Computed:    true,
+			},
+			"http_endpoint": {
+				Type:        types.StringType,
+				Description: "The Elasticsearch resource HTTP endpoint",
+				Computed:    true,
+			},
+			"https_endpoint": {
+				Type:        types.StringType,
+				Description: "The Elasticsearch resource HTTPs endpoint",
+				Computed:    true,
+			},
+
+			"hot":          ElasticsearchTierSchema("'hot' optional topology element", true, "hot"),
+			"coordinating": ElasticsearchTierSchema("'coordinating' optional topology element", false, "coordinating"),
+			"master":       ElasticsearchTierSchema("'master' optional topology element", false, "master"),
+			"warm":         ElasticsearchTierSchema("'warm' optional topology element", false, "warm"),
+			"cold":         ElasticsearchTierSchema("'cold' optional topology element", false, "cold"),
+			"frozen":       ElasticsearchTierSchema("'frozen' optional topology element", false, "frozen"),
+			"ml":           ElasticsearchTierSchema("'ml' optional topology element", false, "ml"),
+
+			"trust_account": ElasticsearchTrustAccountSchema(),
+
+			"trust_external": ElasticsearchTrustExternalSchema(),
+
+			"config": ElasticsearchConfigSchema(),
+
+			"remote_cluster": ElasticsearchRemoteClusterSchema(),
+
+			"snapshot_source": ElasticsearchSnapshotSourceSchema(),
+
+			"extension": ElasticsearchExtensionSchema(),
+
+			"strategy": {
+				Description: "Configuration strategy type " + strings.Join(strategiesList, ", "),
+				Type:        types.StringType,
+				Optional:    true,
+				Validators:  []tfsdk.AttributeValidator{validators.OneOf(strategiesList)},
+			},
+		}),
+	}
+}
+
+func ElasticsearchConfigSchema() tfsdk.Attribute {
+	return tfsdk.Attribute{
+		Description: `Optional Elasticsearch settings which will be applied to all topologies unless overridden on the topology element`,
+		Optional:    true,
+		Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+			// TODO
+			// DiffSuppressFunc: suppressMissingOptionalConfigurationBlock,
```

Comment:
Do the todo?

Reply:
I'm going to remove all `DiffSuppressFunc` TODO comments. 
The Framework doesn't support such functions anymore. 
If `config` can be changed outside of TF, users can resort to `ingore_chages` meta attribute.